zilencer/views.py

```diff
@@ -55,6 +59,20 @@ def validate_bouncer_token_request(entity: Union[UserProfile, RemoteZulipServer]
     validate_token(token, kind)
     return server
 
+def prepare_encrypted_data(devices: List[RemotePushDeviceToken],
+                           payload: EncryptedPayload) -> Tuple[EncryptedData, EncryptedData]:
+    device_map = {d.token: d for d in devices}
+    encrypted_android_data = []
+    encrypted_apple_data = []
+    for token, data in payload:
+        if token in device_map:
+            if device_map[token].kind == RemotePushDeviceToken.GCM:
+                encrypted_android_data.append((device_map[token], data))
+            else:
+                encrypted_apple_data.append((device_map[token], data))
+
+    return encrypted_android_data, encrypted_apple_data
```

Comment:
This code can be made a lot simpler by taking out the `if device_map[token].kind == …` conditional and producing just a single list. Then it can just be a list comprehension, which can just be inlined at the one place it's used.

Then instead of having separate functions `send_android_push_notifications` and `send_apple_push_notifications` which are just simple loops that are very similar to each other, those two can be a single loop, with the `if device.kind == …` conditional inside that loop.